common/persistence/cassandraMetadataPersistence.go

```diff
@@ -176,89 +191,108 @@ func (m *cassandraMetadataPersistence) GetDomain(request *GetDomainRequest) (*Ge
 	var err error
 	info := &DomainInfo{}
 	config := &DomainConfig{}
-	if len(request.ID) > 0 {
-		if len(request.Name) > 0 {
-			return nil, &workflow.BadRequestError{
-				Message: "GetDomain operation failed.  Both ID and Name specified in request.",
-			}
-		}
+	replicationConfig := &DomainReplicationConfig{}
+	var replicationClusters []map[string]interface{}
+	var version int64
 
-		query = m.session.Query(templateGetDomainQuery,
-			request.ID)
-		err = query.Scan(
-			&info.ID,
-			&info.Name,
-			&info.Status,
-			&info.Description,
-			&info.OwnerEmail,
-			&config.Retention,
-			&config.EmitMetric)
-	} else if len(request.Name) > 0 {
-		query = m.session.Query(templateGetDomainByNameQuery,
-			request.Name)
-		err = query.Scan(
-			&info.ID,
-			&info.Name,
-			&info.Status,
-			&info.Description,
-			&info.OwnerEmail,
-			&config.Retention,
-			&config.EmitMetric)
-	} else {
+	if len(request.ID) > 0 && len(request.Name) > 0 {
+		return nil, &workflow.BadRequestError{
+			Message: "GetDomain operation failed.  Both ID and Name specified in request.",
+		}
+	} else if len(request.ID) == 0 && len(request.Name) == 0 {
 		return nil, &workflow.BadRequestError{
 			Message: "GetDomain operation failed.  Both ID and Name are empty.",
 		}
 	}
 
-	if err != nil {
+	handleError := func(name, ID string, err error) error {
+		identity := name
+		if len(ID) > 0 {
+			identity = ID
+		}
 		if err == gocql.ErrNotFound {
-			var d string
-			if len(request.ID) > 0 {
-				d = request.ID
-			} else {
-				d = request.Name
-			}
-
-			return nil, &workflow.EntityNotExistsError{
-				Message: fmt.Sprintf("Domain %s does not exist.", d),
+			return &workflow.EntityNotExistsError{
+				Message: fmt.Sprintf("Domain %s does not exist.", identity),
 			}
 		}
-
-		return nil, &workflow.InternalServiceError{
+		return &workflow.InternalServiceError{
 			Message: fmt.Sprintf("GetDomain operation failed. Error %v", err),
 		}
 	}
 
+	domainName := request.Name
+	if len(request.ID) > 0 {
+		query = m.session.Query(templateGetDomainQuery, request.ID)
+		err = query.Scan(&domainName)
+		if err != nil {
+			return nil, handleError(request.Name, request.ID, err)
+		}
+	}
+
+	query = m.session.Query(templateGetDomainByNameQuery, domainName)
+	err = query.Scan(
+		&info.ID,
+		&info.Name,
+		&info.Status,
+		&info.Description,
+		&info.OwnerEmail,
+		&config.Retention,
+		&config.EmitMetric,
+		&replicationConfig.ActiveClusterName,
+		&replicationConfig.FailoverVersion,
+		&replicationClusters,
+		&version,
+	)
+
+	if err != nil {
+		return nil, handleError(request.Name, request.ID, err)
+	}
+
+	replicationConfig.ActiveClusterName = m.clusterMetadata.GetOrUseDefaultActiveCluster(replicationConfig.ActiveClusterName)
+	for index := range replicationClusters {
```

Comment:
Can you move this out to a helper method.

Reply:
ok